src/main/java/com/nirima/jenkins/plugins/docker/DockerCloud.java

```diff
@@ -727,6 +730,10 @@ protected Object readResolve() {
             dockerApi = new DockerAPI(dockerHost, connectTimeout, readTimeout, version, dockerHostname);
         }
 
+        if (name == null || name.isBlank()) {
+            LOGGER.warn("Docker cloud requires a non-blank name after Jenkins 2.402");
```

Comment:
```suggestion
            LOGGER.warn("Docker cloud requires a non-blank name after Jenkins 2.402");
            return new DockerCloud("docker-" + UUID.randomUUID().toString().substring(0, 8), this);
```

And implement a copy constructor like this (this is what is used in kubernetes-plugin)

```
public DockerCloud(@NonNull String name, @NonNull DockerCloud source) {
        super(name);
        XStream2 xs = new XStream2();
        xs.omitField(Cloud.class, "name");
        xs.unmarshal(XStream2.getDefaultDriver().createReader(new StringReader(xs.toXML(source))), this);
    }
```

Reply:
Thanks very much @Vlatombe !

The XStream2 call caused complaints from Jenkins core that it was trying to marshal unsafe entities, so I switched to a simple assignment in 57d301000787e1bc834c9b8683d68db0ce3276cd .  I'll run some interactive tests to confirm that change is behaving as expected.  Let me know if you detect errors in 57d301000787e1bc834c9b8683d68db0ce3276cd

Reply:
Simple assignment works too, but can be source of maintenance issues because you need to remember that it needs to be updated every time a field is added.